xdsl/transforms/apply_pdl.py

```
@@ -0,0 +1,69 @@
+import os
+from dataclasses import dataclass
+from io import StringIO
+
+from xdsl.context import MLContext
+from xdsl.dialects import builtin, pdl
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters.experimental.pdl import (
+    PDLMatcher,
+    PDLRewriteFunctions,
+)
+from xdsl.ir import Operation, Region, SSAValue
+from xdsl.parser import Parser
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import PatternRewriter
+
+
+@dataclass(frozen=True)
+class ApplyPDLPass(ModulePass):
+    name = "apply-pdl"
+
+    pdl_file: str | None = None
+
+    def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
+        payload_module = op
+        # Target the file containing the PDL specification
+        if self.pdl_file:
+            assert os.path.exists(self.pdl_file)
```

Can you raise an exception here with an error message instead?